src/biz/metrics.go

```diff
@@ -217,29 +230,50 @@ func (mc *MetricsFetcher) memory(mem raw.Memory) Memory {
 	value in memory.stat(see 5.2).
 	However, as the `docker stats` cli tool does, page cache is intentionally
 	excluded to avoid misinterpretation of the output.
-	Also the Swap usage is parsed from memory.memsw.usage_in_bytes, which
+	Also mem.SwapUsage is parsed from memory.memsw.usage_in_bytes, which
 	according to the documentation reports the sum of current memory usage
 	plus swap space used by processes in the cgroup (in bytes). That's why
 	Usage is subtracted from the Swap: to get the actual swap.
 	*/
-	var usage uint64
-	if mem.SwapUsage == 0 { // for systems that have swap disabled
-		usage = mem.RSS
-	} else {
-		usage = mem.RSS + mem.SwapUsage - mem.FuzzUsage
+	var swapOnlyUsage uint64
+	if mem.SwapUsage != 0 { // for systems that have swap disabled
+		swapOnlyUsage = mem.SwapUsage - mem.FuzzUsage
 	}
+	swapUsage := mem.RSS + swapOnlyUsage
 
-	usagePercent := float64(0)
-	if memLimits > 0 {
-		usagePercent = 100 * float64(usage) / float64(memLimits)
+	swapLimit := mem.SwapLimit
+	if mem.SwapLimit > math.MaxInt64/2 {
```

Comment:
this is a nitpick, but I would compare against MaxInt32

Reply:
that would be a different number , this is equivalent to `math.maxUint64/4`  `4294967296 Gb` . But i couldn't find the documentation that mention this on cgroups. But doing test i seen the memory limits exceed this number when is not set inside the cgroup file.